api/proto/teleport/legacy/types/types.proto

```diff
@@ -3145,6 +3145,8 @@ message WebSessionSpecV2 {
   // ConsumedAccessRequestID is the ID of the access request from which additional roles to assume
   // were obtained.
   string ConsumedAccessRequestID = 10 [(gogoproto.jsontag) = "consumed_access_request_id,omitempty"];
+  // Data is arbitrary data within a session object. Considered to be a secret.
+  bytes Data = 11 [(gogoproto.jsontag) = "blob,omitempty"];
```

Comment:
What kind of format does "data" have for SAML use-case? May be not a huge deal but I think whenever you introduce an "arbitrary data" field people tend to start abusing it for storing bunch of random stuff so it eventually becomes hard to understand what's using this field and why.

Any way to make this field more specific to SAML IdP and make it clear what it's expected to contain?

Reply:
https://github.com/crewjam/saml/blob/main/identity_provider.go#L33 This is the object that needs to be stored.

I think it'll be somewhat difficult to make this more structured without introducing a ton of new fields.

Reply:
I think it's fine. I would consider defining a new proto type e.g. `SAMLSessionData` of the same structure (or a subset, depending on what you need) and using it here.

```go
SAMLSessionData SAMLSession = 11 [(gogoproto.jsontag) = "saml_session,omitempty"];
```

You'll have to convert back and forth between it and `saml.Session` in a few places but IMO it would result in a clearer intent.